src/flavors/williams/static/components/input-form.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from "react";
+const cn = require("classnames");
+
+import PrimaryButton from "../../../../base/static/components/ui-elements/primary-button";
+import InputForm from "../../../../base/static/components/input-form";
+import messages from "./messages";
+import constants from "./constants";
+import "./input-form.scss";
+
+const hooks = {
+  postSave: (response, model, defaultPostSave, context) => {
+    if (response.get("location_type") === constants.COMMUNITY_INPUT_CATEGORY_NAME) {
+      // Perform custom postSave behavior if we're submitting a place of the
+      // COMMUNITY_INPUT_CATEGORY_NAME type.
+      context.setState({ stage: 3 });
+    } else {
+      // Otherwise, perform normal routing.
+      defaultPostSave(model);
+    }
+  }
+};
+
+class VVInputForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      stage: 1,
+      mapHasMoved: false
+    };
+    this.onClickContinueForm = this.onClickContinueForm.bind(this);
+    this.onClickExitForm = this.onClickExitForm.bind(this);
+
+    this.props.map.on("dragend", () => {
+      !this.state.mapHasMoved && this.setState({ stage: 2, mapHasMoved: true });
+    });
+
+    const postSaveHookWrapper = (response, model, defaultPostSave) => {
+      return hooks.postSave(response, model, defaultPostSave, this);
+    }
+
```

Comment:
This closure is kind of weird... I tried binding the `postSave` function to the context of `VVInputForm`:
```
hooks.postSave = hooks.postSave.bind(this);
this.props.customHooks["postSave"] = postSave;
```
But when the postSave hook was fired from the parent `InputForm` component, it didn't have the right context. I'm probably forgetting something about the behavior of `this`-bindings here, but the closure takes care of it in any case.